nomenclature/codelist.py

```diff
@@ -191,22 +191,26 @@ def from_directory(
         instance of cls (:class:`CodeList` if not inherited)
 
         """
-        code_list = cls._parse_codelist_dir(path, file_glob_pattern)
 
-        if config is not None:
+        if (
+            config is not None
+            and config.definitions is not None
+            and getattr(config.definitions, path.name) is not None
+            and getattr(getattr(config.definitions, path.name), "repository")
+            is not None
+        ):
             dimension = path.name
-            codelistconfig = getattr(config, dimension, None)
-            if codelistconfig is not None and codelistconfig.repository is not None:
-                repo_path = (
-                    config.repository[codelistconfig.repository].local_path
-                    / codelistconfig.repository_dimension_path
-                )
-                code_list.extend(
-                    cls._parse_codelist_dir(
-                        repo_path,
-                        file_glob_pattern,
-                    )
-                )
+            codelistconfig = getattr(config.definitions, dimension)
+            repo_path = (
+                config.repositories[codelistconfig.repository].local_path
+                / codelistconfig.repository_dimension_path
+            )
+            code_list = cls._parse_codelist_dir(
+                repo_path,
+                file_glob_pattern,
+            ) + cls._parse_codelist_dir(path, file_glob_pattern)
```

Comment:
Even if you keep the if-statement, I'd pull out that line because it's duplicated in both *if* and *else*...

Reply:
Good point, done